### ISSUE TITLE
Vector: modifiers +p and +g not woking 

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#404]
**Full script that generated the error**

```
echo 0.5 1.5 4.5 1.5 | gmt plot -R0/5/0/5 -JX1.75i -W1p -Sv0.2i+s+b+e+gblue+p1p,blue,- -pdf vector
```

**Expected behavior:**
1. vector head filled with blue
2. vector pen attribution is `1p,blue,-`

**Actual behavior:**
![image](https://user-images.githubusercontent.com/3974108/52374624-afa07680-2a2b-11e9-8548-0d6f6ff59860.png)

**System information**

* Operating system: mac
* Version of GMT: 6.0.0 latest

**Comment by PaulWessel** [comment]
There is confusion in the code here and the documentation is unclear.  My recollection of what I meant to implement is this: The **-G**_fill_ and **-W**_pen_ set the default head fill (none) and head outline pen width (none).  However,

- If modifier **+g** is used then head fill is set to default fill (i.e., supplied via **-G**)
- If modifier **+g-** is used then head fill is turned off
- If modifier **+g**_fill_ is given then it sets head fill and overrides any **-G**_fill_ setting
- If modifier **+p** is used then head outline is set to default pen (i.e., supplied via **-W**)
- If modifier **+p-** is used then head outline is turned off
- if modifier **+p**_pen_ is used then the head outline pen is set and overrides any **-W**_pen_

The purpose of **+p** and **+g** is best understood if you imagine reading data files with symbol specifications at the end of each record and that different records may have different vector attribute settings.

However, I think I also wanted to have the capability to use **-W** for the vector stem and then override vector head outline via _+p_ for head outline, i.e., have two separate pens for the same symbol.  I will have to look at this but I think that is a reasonable expectation.

Finally, I also have test examples (e.g., shrinkvec2.sh) that use **-C**_cpt_ to set head fill and/or outline explicitly via the **-W** option's **+c** modifier and there no **-G** or **+g** are actually specified.

Does this explanation make sense?

**Comment by seisman** [comment]
Following code plots vectors with different combinations of -W, -G, +p, +g:
```bash
gmt begin vector png
echo 1 1 4 1 | gmt plot -R0/5/0/2 -Jx1c -Sv0.2i+s+b+e

echo 1 1 4 1 | gmt plot -R0/5/0/2 -Jx1c -Sv0.2i+s+b+e -W1p -Y-1c
echo 1 1 4 1 | gmt plot -R0/5/0/2 -Jx1c -Sv0.2i+s+b+e+p -W1p -Y-1c
echo 1 1 4 1 | gmt plot -R0/5/0/2 -Jx1c -Sv0.2i+s+b+e+p- -W1p -Y-1c
echo 1 1 4 1 | gmt plot -R0/5/0/2 -Jx1c -Sv0.2i+s+b+e+p3p -W1p -Y-1c

echo 1 1 4 1 | gmt plot -R0/5/0/2 -Jx1c -Sv0.2i+s+b+e -Gblack -Y-1c
echo 1 1 4 1 | gmt plot -R0/5/0/2 -Jx1c -Sv0.2i+s+b+e+g -Gblack -Y-1c
echo 1 1 4 1 | gmt plot -R0/5/0/2 -Jx1c -Sv0.2i+s+b+e+g- -Gblack -Y-1c
echo 1 1 4 1 | gmt plot -R0/5/0/2 -Jx1c -Sv0.2i+s+b+e+gred -Gblack -Y-1c

echo 1 1 4 1 | gmt plot -R0/5/0/2 -Jx1c -Sv0.2i+s+b+e -W1p -Gblack -Y-1c
echo 1 1 4 1 | gmt plot -R0/5/0/2 -Jx1c -Sv0.2i+s+b+e+p-+gred -W1p -Gblack -Y-1c
gmt end
```
It seems +p and +g never override -W and -G.
![image](https://user-images.githubusercontent.com/3974108/52484870-5cdad200-2b85-11e9-8c7d-514e2c538415.png)

**Comment by joa-quim** [comment]
Yes, that was my experience as well when writing the Julia wrapper. +g was always ignored. I mentioned it in a couple of issues but never opened one dedicated to that. Like this one, thanks.

**Comment by PaulWessel** [comment]
O know (and know why).  I just wanted to get feedback that what I oulined above should be the expectation.  Do @seisman and @joa-quim agree?

**Comment by seisman** [comment]
@PaulWessel Yes, that's exactly what I expect.

**Comment by joa-quim** [comment]
I find it a bit confusing. Why would I do **-Gred +g-**? **-G** is generic for psxy filling, so for me **+g**fill and **-G**fill are just synonyms, where latter will work for GMT4 vector syntax as well.

**Comment by PaulWessel** [comment]
You may have a file with 50 points and 5 of them you want not to fill but the rest should be red.  So you use **-G**_red_ on the command line but for 5 of the records you have **+g-** to turn it off for those symbols (i.e., every records ends with the symbol code and modifiers.  Yes, on the command line, **+g**_fill_ would/should be synonymous with **-G**_fill_, but in the file it is "per record".

**Comment by seisman** [comment]
Closed by #415.
